gen/gen.go

```diff
@@ -69,31 +84,45 @@ func Gen(exercise string, j interface{}, t *template.Template) error {
 	if dirMetadata == "" {
 		return errors.New("unable to determine current path")
 	}
+
 	jFile := filepath.Join("exercises", exercise, "canonical-data.json")
 	// find and read the json source file
-	jPath, jOrigin, jCommit := getPath(jFile)
-	jSrc, err := ioutil.ReadFile(filepath.Join(jPath, jFile))
-	if err != nil {
-		return err
+	jPath, jOrigin, jCommit := getLocal(jFile)
```

Comment:
I think the `getLocal` function could be refined a bit, moving `jFile` into it and returning the `ioutil.Readfile` in line 101. It then needs a different way to decide if there is a valid local source to use. :thinking: